admin/server/organizations.go

```diff
@@ -212,6 +213,12 @@ func (s *Server) AddOrganizationMember(ctx context.Context, req *adminv1.AddOrga
 		return nil, status.Error(codes.PermissionDenied, "not allowed to add org members")
 	}
 
+	// Validate email address
+	_, err = mail.ParseAddress(req.Email)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid email address: %q", req.Email)
+	}
+
```

Comment:
If you refactor `InviteUserToOrganization` to accept an `opts` parameter instead, you can add a validation tag for `email` there instead (see `database.InsertProjectOptions` as an example).

Alternatively (or additionally), you can use the validation tags in the proto file to enforce it at the request parsing stage.

Reply:
validating before call to `InviteUserToOrganization` will prevent from doing extra work like fetching org and role from db. proto file already has `[(validate.rules).string.email = true]` so I think we are good, we can remove the check from here.